services/QuillLMS/spec/controllers/teachers/classroom_manager_controller_spec.rb

```diff
@@ -341,19 +341,23 @@
 
   describe '#scorebook' do
     let(:teacher) { create(:teacher) }
-    let(:classroom) { create(:classroom) }
     let(:classroom1) { create(:classroom) }
+    let(:classroom2) { create(:classroom) }
+    let(:classroom3) { create(:classroom) }
+    let(:classrooms_teacher1) {create(:classrooms_teacher, user_id: teacher.id, classroom_id: classroom1.id, order: 1)}
+    let(:classrooms_teacher2) {create(:classrooms_teacher, user_id: teacher.id, classroom_id: classroom2.id, order: 0)}
+    let(:classrooms_teacher3) {create(:classrooms_teacher, user_id: teacher.id, classroom_id: classroom3.id, order: 2)}
 
     before do
       allow(controller).to receive(:current_user) { teacher }
-      allow(RawSqlRunner).to receive(:execute).and_return([classroom, classroom1])
+      allow(RawSqlRunner).to receive(:execute).and_return([classroom2, classroom1, classroom3])
       allow(controller).to receive(:classroom_teacher!) { true }
     end
 
     context 'when classroom id is passed' do
-      it 'should assign the classrooms and classroom' do
+      it 'should assign the classrooms (sorted by order) and classroom' do
         get :scorebook, params: { classroom_id: classroom1.id }
-        expect(assigns(:classrooms)).to eq ([classroom, classroom1].as_json)
+        expect(assigns(:classrooms)).to eq ([classroom2, classroom1, classroom3].as_json)
```

Comment:
By setting your return value in an `allow` call, and then checking that you got it back, you're not actually testing that your code change works.  That is, even if your `ORDER BY` clause is broken somehow, this test won't be able to tell.  You might just be able to take out that `allow(RawSqlRunner)` call entirely and the tests should pass?

Reply:
just following up on our chat-- since we agreed this entire test file is pretty old and non-trivial to refactor, I wrote up a card to refactor [here](https://www.notion.so/quill/Refactor-ClassroomManagerController-spec-file-f3b8f1327ca5491fb619a37ed4d3812a).